kratos/operations/operation.h

```diff
@@ -0,0 +1,191 @@
+    }
+
+    /**
+     * @brief Execute method is used to execute the Operation algorithms.
+     */
+    virtual void Execute()
+    {
+        KRATOS_ERROR << "Calling base class Execute. Please override this method in the corresonding Operation" << std::endl;
+    }
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     */
+    virtual const Parameters GetDefaultParameters() const
+    {
+        KRATOS_ERROR << "Calling the base Operation class GetDefaultParameters. Please implement the GetDefaultParameters in your derived process class." << std::endl;
+        const Parameters default_parameters = Parameters(R"({})" );
+
+        return default_parameters;
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const
+    {
+        return "Operation";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const
+    {
+        rOStream << "Operation";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const
+    {
+    }
+
+    ///@}
+}; // Class Operation
+
+///@name Type Definitions
+///@{
+
+#ifndef KRATOS_REGISTER_OPERATION
+#define KRATOS_REGISTER_OPERATION(                                                                              \
+    module_name,                                                                                                \
+    operation_name,                                                                                             \
+    operation_prototype)                                                                                        \
+    {                                                                                                           \
+        std::string all_path = std::string("Operations.All.") + operation_name;                                 \
+        if (!Registry::HasItem(all_path)) {                                                                     \
+            auto& r_operation_item = Registry::AddItem<RegistryItem>(all_path);                                 \
+            r_operation_item.AddItem<RegistryValueItem<Operation>>("Prototype", operation_prototype);           \
+        } else {                                                                                                \
+            KRATOS_ERROR << "Operation '" << operation_name << "' is already registered." << std::endl;         \
+        }                                                                                                       \
+        std::string module_path = std::string("Operations.") + module_name + std::string(".") + operation_name; \
+        if (!Registry::HasItem(module_path)) {                                                                  \
```

Comment:
Shall we do this check? or we should leave it to registry to send an error?

Reply:
(see above answer)

Reply:
But here you are avoiding the error. Is it intentional?